eppo_core/src/background/thread.rs

```diff
@@ -0,0 +1,74 @@
+use super::runtime::BackgroundRuntime;
+
+/// An owning handle to a background thread running tokio runtime.
+///
+/// When the handle is dropped, the tokio runtime is commanded to exit and the thread shuts down.
+pub struct BackgroundThread {
+    join_handle: std::thread::JoinHandle<()>,
+    runtime: BackgroundRuntime,
+}
```

Comment:
Background runtime does not own the tokio runtime and it needs something else to actually drive tokio runtime. Background thread is the thing that creates tokio runtime and runs it on a native OS thread. (There will be other ways to run tokio in the future.)

Reply:
> Background runtime does not own the tokio runtime

Not sure I understand this, I see that `BackgroundRuntime` owns a `tokio_runtime: tokio::runtime::Handle`

Reply:
[`Handle`](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html) is a non-owning reference to a tokio runtime. [`Runtime`](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html) is the actual thing

Reply:
ah gotcha